src/danfojs-base/core/frame.ts

```diff
@@ -2777,8 +2781,9 @@ export default class DataFrame extends NDframe implements DataFrameInterface {
             const colsAdded: string[] = [];
             const newColumns = this.columns.map(col => {
                 if (mapper[col] !== undefined) {
-                    colsAdded.push(mapper[col]);
-                    return mapper[col]
+                    const newCol = `${mapper[col]}`;
```

Comment:
Hey @Devwulf I'm not sure why you did this, can you explain?

Reply:
> Hey @Devwulf I'm not sure why you did this, can you explain?

So with the new mapper type, you can type in either a `string` or a `number` as the value in a mapping pair. Since the dataframe columns array only accept strings, line 2784 basically converts each mapped value as strings. This is so they can be pushed into the `colsAdded` array and so that `newColumns` stay as a `string[]` instead of being a `(string | number)[]` array.

Perhaps this function can also be separated into two with the only difference being the axis value they take in, but you might as well start doing this to all functions that take in an `axis` option, which might complicate things. This is what I'm thinking about:

```js
rename(
    mapper: {
        [index: string]: string
    }, 
    options?: {
        axis?: 1, // this function is returned if axis is 1
        inplace?: boolean
    }
): DataFrame | void {
    // Rename the columns
}

rename(
    mapper: {
        [index: string | number]: string | number
    }, 
    options?: {
        axis?: 0, // this function is returned if axis is 0
        inplace?: boolean
    }
): DataFrame | void {
    // Rename the row indexes
}
```

It might be convenient to do something similar as well for the `inplace` option, so that the function returns the proper values (either void or DataFrame, not both) depending on the `inplace` option, since I find myself always casting to DataFrame every time `inplace` is false.

Reply:
> > Hey @Devwulf I'm not sure why you did this, can you explain?
> 
> So with the new mapper type, you can type in either a `string` or a `number` as the value in a mapping pair. Since the dataframe columns array only accept strings, line 2784 basically converts each mapped value as strings. This is so they can be pushed into the `colsAdded` array and so that `newColumns` stay as a `string[]` instead of being a `(string | number)[]` array.
> 
> Perhaps this function can also be separated into two with the only difference being the axis value they take in, but you might as well start doing this to all functions that take in an `axis` option, which might complicate things. This is what I'm thinking about:
> 
> ```js
> rename(
>     mapper: {
>         [index: string]: string
>     }, 
>     options?: {
>         axis?: 1, // this function is returned if axis is 1
>         inplace?: boolean
>     }
> ): DataFrame | void {
>     // Rename the columns
> }
> 
> rename(
>     mapper: {
>         [index: string | number]: string | number
>     }, 
>     options?: {
>         axis?: 0, // this function is returned if axis is 0
>         inplace?: boolean
>     }
> ): DataFrame | void {
>     // Rename the row indexes
> }
> ```
> 
> It might be convenient to do something similar as well for the `inplace` option, so that the function returns the proper values (either void or DataFrame, not both) depending on the `inplace` option, since I find myself always casting to DataFrame every time `inplace` is false.

Oh my bad, I see why now. Thanks for taking out time to explain, and yes you're correct, columns only take in string values.